### ISSUE TITLE
After replacing the Muon optimizer, the val loss and MAP behavior are inconsistent.

### ISSUE THREAD

**Issue by HuberyQ** [ultralytics/ultralytics#22642]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello.
I recently replaced the optimizer with Muon+AdamW, hoping to speed up training YOLO12. Training did finish faster, and the loss curve (green) was better than the default SGD (pink). However, **I observed that while the val loss was continuously decreasing, the MAP was decreasing**, which is somewhat strange. And the output model did not perform better on the test dataset.Why is this happening?
Thank you for your help!

<img width="2063" height="1286" alt="Image" src="https://github.com/user-attachments/assets/4a34ef0c-5aff-4792-a67c-c2be133fd277" />

### Additional

**Optimizer-related parameters:**
Muon lr=0.001, momentum=0.9, weight_decay=0.0005, for hidden layer 2D weights;
AdamW lr=0.000714, betas=(0.9, 0.999), weight_decay=0.0 for BN, head..

**Comment by UltralyticsAssistant** [comment]
👋 Hello @HuberyQ, thank you for your interest in Ultralytics 🚀! This is an automated response to help get things started, and an Ultralytics engineer will follow up with you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
- Include the exact training/validation commands, full logs, and plots (loss and metrics)📈
- Share your environment details (OS, Python, PyTorch, CUDA versions) and `ultralytics` version
- Provide the model YAML, data YAML, seeds, image sizes, and any callbacks or custom settings
- Since you’re using custom optimizers, include the optimizer config/code snippet you used and how it’s integrated

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Details that help:
- Which model variant you trained and with what image size, batch size, and epochs
- Any scheduler settings, EMA, gradient clipping, AMP, or multi-scale
- How validation was performed (same data split, conf/iou thresholds, NMS settings)
- Reproducible comparison runs between default optimizer and your custom setup

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Thanks for the detailed comparison—what you’re seeing can occur when the validation loss (box/cls/dfl) optimizes differently than the evaluation metric, so a lower val loss can coincide with worse mAP, especially after changing optimizers and parameter groups; see the brief distinction in [Loss vs. evaluation metrics](https://www.ultralytics.com/glossary/loss-function) and how mAP is computed in the [Performance metrics guide](https://docs.ultralytics.com/guides/yolo-performance-metrics/). As a clean baseline, please compare against YOLO11 using the built-in optimizers only (`optimizer='auto'`, `SGD`, `AdamW`); YOLO12 is known to be training-unstable. For reproducibility, re-evaluate both weights on the same data and attach artifacts:

```python
from ultralytics import YOLO

for w in ["runs/detect/train_sgd/weights/best.pt", "runs/detect/train_muon/weights/best.pt"]:
    YOLO(w).val(data="data.yaml", plots=True)
```

Also share: `yolo checks` output (text), both runs’ `results.csv`, `PR_curve.png` and `confusion_matrix.png`, plus the exact optimizer diff you applied (Muon isn’t among the supported choices in [`build_optimizer`](https://docs.ultralytics.com/reference/engine/trainer/#build_optimizer)). If the metrics divergence persists on YOLO11 with stock optimizers, we’ll dig in further.
